File: 0066-plus-one/0066-plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
const plusOne = digits => {
    let carry = 0;
    for (let i = digits.length - 1; i >= 0; i--) {
        if (digits[i] === 9) {
            carry = 1;
            if (i === 0 && carry === 1) {
                digits[i] = 0;
                digits.unshift(1);
            } else {
                digits[i] = 0;
            }
        } else if (i === digits.length - 1) {
            digits[i]++;
            break;
        } else {
            digits[i] += carry;
            break;
        }
    }
    return digits;
};

File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 0
        for i in range(len(digits) - 1, -1, -1):
            digit = digits[i]
            if i == 0 and digit == 9:
                digits[i] = 0
                digits = [1] + digits
            elif i != 0 and digit == 9:
                carry = 1
                digits[i] = 0
            elif i == len(digits) - 1:
                digits[i] += 1
                break
            elif i != len(digits) - 1:
                digits[i] += carry
                break
        return digits